src/components/table/expandable/cell/index.tsx

```diff
@@ -1,51 +1,79 @@
-import { Component, ComponentDidLoad, Element, Event, EventEmitter, h, Host, Listen, Prop, Watch } from "@stencil/core"
+import {
+	Component,
+	ComponentWillLoad,
+	Element,
+	Event,
+	EventEmitter,
+	h,
+	Host,
+	Listen,
+	Prop,
+	State,
+	Watch,
+} from "@stencil/core"
 
 @Component({
 	tag: "smoothly-table-expandable-cell",
 	styleUrl: "style.css",
 	scoped: true,
 })
-export class TableExpandableCell implements ComponentDidLoad {
+export class TableExpandableCell implements ComponentWillLoad {
+	private expansionElement?: HTMLTableRowElement
+	private beginOpen: boolean
 	@Element() element: HTMLSmoothlyTableExpandableCellElement
-	expansionElement?: HTMLTableRowElement
-	@Event() expansionOpen: EventEmitter<HTMLElement>
-	@Event() expansionLoad: EventEmitter<void>
+	@State() allowSpotlight = true
```

Comment:
why we need state `allowSpotlight`?  we dont even change it inside this component? you just @Watch it and emit to the event `expandableLoad` (is this event needed too?
for my understanding, what we want is just to show the spotlight or not, then why not change spotlight to a Prop()?